source/code/plugin/health/health_monitor_utils.rb

```diff
@@ -0,0 +1,369 @@
+                        else
+                            owner_kind = pod['kind']
+                            controller_name = pod['metadata']['name']
+                            #log.info "#{JSON.pretty_generate(pod)}"
+                        end
+
+                        namespace = pod['metadata']['namespace']
+                        status = pod['status']['phase']
+
+                        workload_name = ''
+                        if owner_kind.nil?
+                            owner_kind = 'Pod'
+                        end
+                        case owner_kind.downcase
+                        when 'job'
+                            # we are excluding jobs
+                            next
+                        when 'replicaset'
+                            # get the labels, and see if there is a match. If there is, it is the deployment. If not, use replica set name/controller name
+                            labels = pod['metadata']['labels'].to_h
+                            labels.each {|k,v|
+                                lookup_key = "#{namespace}-#{k}=#{v}"
+                                if deployment_lookup.key?(lookup_key)
+                                    workload_name = deployment_lookup[lookup_key]
+                                    break
+                                end
+                            }
+                            if workload_name.empty?
+                                workload_name = "#{namespace}~~#{controller_name}"
+                            end
+                        when 'daemonset'
+                            workload_name = "#{namespace}~~#{controller_name}"
+                        else
+                            workload_name = "#{namespace}~~#{pod['metadata']['name']}"
+                        end
+
+                        if pods_ready_percentage_hash.key?(workload_name)
+                            total_pods = pods_ready_percentage_hash[workload_name]['totalPods']
+                            pods_ready = pods_ready_percentage_hash[workload_name]['podsReady']
+                        else
+                            total_pods = 0
+                            pods_ready = 0
+                        end
+
+                        total_pods += 1
+                        if status == 'Running'
+                            pods_ready += 1
+                        end
+
+                        pods_ready_percentage_hash[workload_name] = {'totalPods' => total_pods, 'podsReady' => pods_ready, 'namespace' => namespace, 'workload_name' => workload_name, 'kind' => owner_kind}
+                    rescue => e
+                        log.info "Error when processing pod #{pod['metadata']['name']} #{e.message}"
+                    end
+                end
+                return pods_ready_percentage_hash
+            end
+
+            def get_node_state_from_node_conditions(node_conditions)
```

Comment:
I am assuming we tested this for all node conditions ? Do you have a cluster to look at ?

Reply:
yeah. dilipr-health-test